caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,12 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+// calculateEtag produces a strong etag by default. Prefix the result with "W/" to convert this into a weak one.
+// see https://tools.ietf.org/html/rfc7232#section-2.3
+func calculateEtag(d os.FileInfo) string {
+	return fmt.Sprintf(`"%x-%x"`, d.ModTime().Unix(), d.Size())
```

Comment:
I just checked Nginx 1.10.1:

* when `Accept-Encoding:gzip` is sent and *gzip_static* is on, the precompressed content is served if available.
* when `Accept-Encoding:gzip` is missing or *gzip_static* is off or the precompressed content is not available, the uncompressed content is served.
* the ETag is computed from the timestamp+length algorithm in both cases above.
* strong ETags are always used.